io/driver/module_unload_load.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+CONFIG_FILE="$AVOCADO_TEST_DATADIR"/config
+DRIVER=`lspci -k | grep -iw "Kernel driver in use" | cut -d ':' -f2 | sort | uniq`
+
+#Loads the kernel driver modules
+module_load() {
+    echo "Reloading driver $1"
+    modprobe $1
+    if [ $? != 0 ]; then
+        echo "Failed to load driver module $1"
+        break;
+    fi
+    echo
+}
+
+#Unloads the kernel driver modules
+module_unload() {
+    for i in $(cat $CONFIG_FILE | grep "$1=" | awk -F'=' '{print $2}'); do
+        module_unload $i
+        if [ $? != 0 ]; then
+            echo "Failed to unload driver module $i"
+            return
+        fi
+    done
+    echo "Unloaded driver $1"
+    rmmod $1
```

Comment:
As for virtio-pci, I ran the code on a PowerKVM guest and it worked fine. Please can you let me know on which system and where did it hang.